### PR TITLE
KAFKA-13846: Use the new addMetricsIfAbsent API

### PR DESCRIPTION
**Pull request by guozhangwang**
Use the newly added function to replace the old `addMetric` function that may throw illegal argument exceptions.

Although in some cases concurrency should not be possible they do not necessarily remain always true in the future, so it's better to use the new API just to be less error-prone.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
cc @vamossagar12 @cadonna

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java [hunk 1]
```diff
@@ -502,9 +502,8 @@ public <T> void addStoreLevelMutableMetric(final String taskId,
             storeLevelTagMap(taskId, metricsScope, storeName)
         );
         if (metrics.metric(metricName) == null) {
-            final MetricConfig metricConfig = new MetricConfig().recordLevel(recordingLevel);
+            metrics.addMetricIfAbsent(metricName, new MetricConfig().recordLevel(recordingLevel), valueProvider);
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java:505]: Just take here as an example: here we depend on two threads would not try to create the same store-name and the same task-id but that may not always hold true in the future, so better using the newly introduced API. cc @cadonna .

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java:505]: Agreed!

**Review by cadonna — APPROVED** [review]
@guozhangwang Thank you for the PR!

LGTM!

**Comment by cadonna** [comment]
@guozhangwang the test failures are related.

**Comment by cadonna** [comment]
You need to replace 
```
metrics.addMetric(eq(metricName), eqMetricConfig(metricConfig), eq(VALUE_PROVIDER));
```
with

```
expect(metrics.addMetricIfAbsent(eq(metricName), eqMetricConfig(metricConfig), eq(VALUE_PROVIDER))).andReturn(null);
```

in `StreamsMetricsImplTest#shouldAddNewStoreLevelMutableMetric()`

**Review by Kvicii — APPROVED** [review]

**Pull request merged**